src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java

```diff
@@ -0,0 +1,138 @@
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * The SunbirdVCIServiceImpl.
+ *
+ * @author Dhanendra
+ */
+@Service
+public class SunbirdVCIServiceImpl implements SunbirdVCIService {
+
+    Logger logger = DigitalCardRepoLogger.getLogger(SunbirdVCIServiceImpl.class);
+
+    @Autowired
+    private TokenIDGenerator tokenIDGenerator;
+    @Value("${mosip.digitalcard.sunbird.vci.registry.url}")
+    private String registryURL;
+
+    @Value("${mosip.digitalcard.sunbird.vci.search.registry.url}")
+    private String searchRegistryURL;
+
+    @Value("${mosip.digitalcard.sunbird.vci.policy.Name}")
+    private String policyName;
+
+    @Value("${mosip.digitalcard.sunbird.vci.policy.expiry}")
+    private String policyExpiry;
+
+    @Value("${mosip.digitalcard.sunbird.vci.policy.issuedOn}")
+    private String policyIssuedOn;
+
+    @Value("${mosip.digitalcard.sunbird.vci.policy.id}")
+    private String policyId;
+    @Value("${mosip.digitalcard.sunbird.vci.request.benefits}")
+    private List<String> sunbirdVciBenefits;
+
+    @Autowired
+    RestClient restClient;
+
+    @Override
+    public void createRegistry(Map<String,Object> identity) {
+        CreateInsuranceDto createInsuranceDto=new CreateInsuranceDto();
+            try {
+                createIssuanceRequest(identity,createInsuranceDto);
+                String response=restClient.postApi(registryURL,null,"","", MediaType.APPLICATION_JSON,createInsuranceDto,String.class);
+                identity.put("policyNumber",createInsuranceDto.getPolicyNumber());
+                logger.info("Created Registry for RID: {}",identity.get("RID"));
+            }  catch (Exception e) {
+                logger.error("Failed Creating Registry for RID: {}",identity.get("RID"));
+                throw new DigitalCardServiceException(e);
+            }
+    }
+
+    @Override
+    public String searchPolicy(String policyNumber) {
+        RegistrySearchRequestDto registrySearchRequestDto=new RegistrySearchRequestDto();
+
+        Map<String, Map<String, String>> filters=new HashMap<>();
+        Map<String,String> filter=new HashMap<>();
+        try {
+            registrySearchRequestDto.setOffset(0);
+            registrySearchRequestDto.setLimit(1);
+            filter.put("eq",policyNumber);
+            filters.put("policyNumber",filter);
+            registrySearchRequestDto.setFilters(filters);
+            List<String> pathsegments = null;
+            String response=restClient.postApi(searchRegistryURL,null,"","", MediaType.APPLICATION_JSON,registrySearchRequestDto,String.class);
+            if(response.contains("osid")){
+                searchPolicy(generatePolicyNumber(policyId));
```

Comment:
This goes into a infinite resursive loop, possible to break it after 15 (config) times ?

Reply:
taken care

Reply:
This implementation is not correct. inside the while loop we need to generate an id and check again. but we are looping the same result those many times.

Reply:
taken care